rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.allocation_explain/10_basic.yml

```diff
@@ -164,19 +386,674 @@
+      cluster.allocation_explain:
+        index: "test"
+        shard: 1.0
+        primary: true
+
+---
+"cluster shard allocation explanation test with numerical primary parameter passed in URL":
+  - requires:
+      capabilities:
+        - method: GET
+          path: /_cluster/allocation/explain
+          capabilities: [ query_parameter_support ]
+      test_runner_features: [ capabilities ]
+      reason: "Query parameter support was added in version 9.2.0"
+
+  - do:
+      indices.create:
+        index: test
+
+  - match: { acknowledged: true }
+
+  - do:
+      catch: /illegal_argument_exception/
+      cluster.allocation_explain:
+        index: "test"
+        shard: 0
+        primary: 0
+
+---
+"cluster shard allocation explanation test with invalid primary parameter passed in URL":
+  - requires:
+      capabilities:
+        - method: GET
+          path: /_cluster/allocation/explain
+          capabilities: [ query_parameter_support ]
+      test_runner_features: [ capabilities ]
+      reason: "Query parameter support was added in version 9.2.0"
+
+  - do:
+      indices.create:
+        index: test
+
+  - match: { acknowledged: true }
+
+  - do:
+      catch: /illegal_argument_exception/
+      cluster.allocation_explain:
+        index: "test"
+        shard: 0
+        primary: truee
+
+---
+"cluster shard allocation explanation test with a valid, string primary parameter passed in URL":
+  - requires:
+      capabilities:
+        - method: GET
+          path: /_cluster/allocation/explain
+          capabilities: [ query_parameter_support ]
+      test_runner_features: [ capabilities ]
+      reason: "Query parameter support was added in version 9.2.0"
+
+  - do:
+      indices.create:
+        index: test
+
+  - match: { acknowledged: true }
+
+  - do:
+      cluster.allocation_explain:
+        index: "test"
+        shard: 0
+        primary: "true"
+
+  - match: { current_state: "started" }
+  - is_true: current_node.id
+  - match: { index: "test" }
+  - match: { shard: 0 }
+  - match: { primary: true }
+  - is_true: can_remain_on_current_node
+  - is_true: can_rebalance_cluster
+  - is_true: can_rebalance_to_other_node
+  - is_true: rebalance_explanation
+
+---
+"cluster shard allocation explanation test with an invalid, string primary parameter passed in URL":
+  - requires:
+      capabilities:
+        - method: GET
+          path: /_cluster/allocation/explain
+          capabilities: [ query_parameter_support ]
+      test_runner_features: [ capabilities ]
+      reason: "Query parameter support was added in version 9.2.0"
+
+  - do:
+      indices.create:
+        index: test
+
+  - match: { acknowledged: true }
+
+  - do:
+      catch: /illegal_argument_exception/
+      cluster.allocation_explain:
+        index: "test"
+        shard: 0
+        primary: "truee"
+
+---
+"cluster shard allocation explanation test with numerical current node parameter passed in URL":
```

Comment:
This test is kinda misleading - there's nothing invalid about `current_node: 1` as the title suggests, the problem is the `primary: 0`, and we're already testing that elsewhere.